nodejs/week3/homework/warmup/index.js

```diff
@@ -0,0 +1,118 @@
+      req.query.secondNumber
+        .map((secValue) => secValue)
+        .reduce((valueOne, valueTwo) => Number(valueOne) + Number(valueTwo)) +
+      Number(firstValue);
+    res.send(`${result}`);
+  }
+  res.send(`${Number(firstValue) + Number(secValue)}`);
+});
+
+//subtraction
+app.get("/calculator/subtraction", (req, res) => {
+  const firstValue = Number(req.query.firstNumber);
+  const secValue = Number(req.query.secondNumber);
+  console.log(typeof secValue);
+  if (Array.isArray(req.query.secondNumber)) {
+    const result = req.query.secondNumber
+      .map((secValue) => secValue)
+      .reduce((valueOne, valueTwo) => Number(valueOne) + Number(valueTwo));
+
+    res.json(firstValue - result);
+  }
+  res.json(firstValue - secValue);
+});
+
+//multiply
+app.get("/calculator/multiply", (req, res) => {
+  const firstValue = Number(req.query.firstNumber);
+  const secValue = Number(req.query.secondNumber);
+  console.log(typeof secValue);
+  if (Array.isArray(req.query.secondNumber)) {
+    const result = req.query.secondNumber
+      .map((secValue) => secValue)
+      .reduce((valueOne, valueTwo) => Number(valueOne) * Number(valueTwo));
+
+    res.json(firstValue * result);
+  }
+  res.json(firstValue * secValue);
+});
+//division
+app.get("/calculator/division", (req, res) => {
+  const firstValue = Number(req.query.firstNumber);
+  const secValue = Number(req.query.secondNumber);
+  console.log(typeof secValue);
+  if (Array.isArray(req.query.secondNumber)) {
+    const result = req.query.secondNumber
+      .map((secValue) => secValue)
+      .reduce((valueOne, valueTwo) => Number(valueOne) / Number(valueTwo));
+
+    res.json(firstValue / result);
+  }
+  res.json(firstValue / secValue);
+});
+//post division
+app.post("/calculator/division", (req, res) => {
+  const firstValue = Number(req.body.firstNumber);
+  const secValue = Number(req.body.secondNumber);
+  console.log(typeof secValue);
+  if (Array.isArray(req.body.secondNumber)) {
+    const result = req.body.secondNumber
+      .map((secValue) => secValue)
+      .reduce((valueOne, valueTwo) => Number(valueOne) / Number(valueTwo));
+
+    res.json(firstValue / result);
+  }
+  res.json(firstValue / secValue);
+});
+
+//post multiply
+app.post("/calculator/multiply", (req, res) => {
+  const firstValue = Number(req.body.firstNumber);
+  const secValue = Number(req.body.secondNumber);
+  console.log(typeof secValue);
+  if (Array.isArray(req.body.secondNumber)) {
+    const result = req.body.secondNumber
+      .map((secValue) => secValue)
+      .reduce((valueOne, valueTwo) => Number(valueOne) * Number(valueTwo));
+
+    res.json(firstValue * result);
+  }
+  res.json(firstValue * secValue);
+});
+
+//another method
+//Getting data through the request body using POST
+//multiply
+app.post("/calculator/multiply", (req, res) => {
+  if (!req.query.firstNumber || !req.query.secondNumber) {
+    res.status(400).send("provide correct information");
+    return;
+  }
```

Comment:
This route doesn't seem right, maybe you forgot it here?
You have the POST multiply route in line 85.
Also, here you're getting the parameters from req.query while you should get them from req.body since this is a POST request. The route in line 85 does that correctly.